src/rust/engine/watch/src/lib.rs

```diff
@@ -96,39 +104,56 @@ impl InvalidationWatcher {
       }
     }
 
+    Ok(Arc::new(InvalidationWatcher(Mutex::new(Inner {
+      watcher,
+      executor,
+      liveness: liveness_receiver,
+      enabled,
+      background_task_inputs: Some((
+        ignorer,
+        canonical_build_root,
+        liveness_sender,
+        watch_receiver,
+      )),
+    }))))
+  }
+
+  ///
+  /// Starts the background task that monitors watch events. Panics if called more than once.
+  ///
+  pub fn start<I: Invalidatable>(&self, invalidatable: &Arc<I>) {
+    let mut inner = self.0.lock();
+    let (ignorer, canonical_build_root, liveness_sender, watch_receiver) = inner
+      .background_task_inputs
+      .take()
```

Comment:
This moves the contents out of the Option right?

Reply:
Yep. The next line asserts that `start` is not called more than once.